### ISSUE TITLE
Recent bug in torch.cat() on Variables?

### ISSUE THREAD

**Issue by fritzo** [pytorch/pytorch#3452]
As of 66d24c5 I'm seeing errors when calling `torch.cat()`
```py
torch.cat(my_difficult_to_reproduce_list_of_torch_variables)
```
```
RuntimeError: dimension specified as 0 but tensor has no dimensions
```
As a workaround, I'm using `torch.stack(-).squeeze()`.

**Comment by zou3519** [comment]
I'm on master and the following works for me:
![image](https://user-images.githubusercontent.com/5652049/32347632-9bab4664-bfe7-11e7-81a5-ec8c2651217a.png)

Could you post a minimal example that triggers the error?

**Comment by fritzo** [comment]
@zou3519 Thanks for the quick response. I have not been able to reproduce with a small example. Do you have any tips for a workaround to torch's lack of support for `repr()` to help me generate an example?

**Comment by fritzo** [comment]
```py
log_weights = huge_function_in_private_repo()
print(log_weights)
# [Variable containing:
# -89.9156
# [torch.DoubleTensor of size 1]
# , Variable containing:
# -64.7077
# [torch.DoubleTensor of size 1]
# ]
torch.cat(log_weights)
# *** RuntimeError: dimension specified as 0 but tensor has no dimensions
```

**Comment by gchanan** [comment]
is this https://github.com/pytorch/pytorch/issues/1530?

**Comment by colesbury** [comment]
I suspect one of the tensors is zero-dimensional at the ATen level, which causes wrap_dim to fail.

**Comment by colesbury** [comment]
@fritzo this should be fixed in `master` (and `v0.3.0`). Can you check that you no longer get this error?

**Comment by fritzo** [comment]
@colesbury Yes, I can confirm that I no longer get this error in pytorch master as of dc10083

**Comment by fritzo** [comment]
Great, this is also fixed on the `v0.3.0` branch as of 9a67882

**Comment by jiapei100** [comment]
I met the same issue at line 78 of [https://github.com/hunkim/PyTorchZeroToAll/blob/master/12_2_hello_rnn.py](https://github.com/hunkim/PyTorchZeroToAll/blob/master/12_2_hello_rnn.py) . Can anybody help to have a look? I'm using torch 0.4.0 and torchvision 0.2.1 ...

> >>> import torch
> >>> torch.__version__
> '0.4.0'
> >>> import torchvision
> >>> torchvision.__version__
> '0.2.1'

Cheers
Pei
